### DISCUSSION TITLE
Is there a reason to call Node "ASTNode" in the Ecore metamodel?

### DISCUSSION THREAD

**Discussion by alessiostalla** [Strumenta/kolasu#122] [Ideas]
As the title says. Given that we're going to finalize the Kolasu v2 (aka *Lasu?) metamodel, and we're thinking about using it as the starting point for other *Lasu implementations, it strikes me as odd that all the classes are named the same as their Kolasu counterpart except for Node which is called ASTNode in the Ecore metamodel. If we use a generator to write a, say, Python implementation of the metamodel, it would contain an ASTNode class instead of a Node class.

**Comment by ftomassetti** [comment]
Regarding the metamodel, maybe it should be named StarLasu and not Kolasu, as it is going to be shared across the StarLasu libraries.

Regarding ASTNode, I think this is a good point and we should probably change the name in the metamodel or rename the class (perhaps defining a type alias to facilitate the transition).

I can think of two factors to consider, both not so strong:
1. I thought that we could potentially have in the future both the concept of Node and ASTNode as a specialized of Node. I am not sure this makes sense, perhaps it would make sense to distinguish between ASTNodes in projectional editors and in textual editors, but in that case both would be "ASTNodes"
2. Node is a very generic term and for example XML parsers use the term Node to indicate an element of the XML tree, so maybe picking a more specific term could avoid confusion

**Reply by alessiostalla** [reply]
I'm not against ASTNode per se but I find it slightly problematic that the name in the metamodel differs from the name in the code.
Anyway, regarding your points let's not forget that we have packages/namespaces so the actual name is e.g. `com.strumenta.kolasu.model.Node` in Kotlin.

**Reply by ftomassetti** [reply]
Should we then rename the "Node" class into ASTNode then?

**Reply by alessiostalla** [reply]
I would rather rename ASTNode to Node to limit the impact on existing projects...

**Reply by ftomassetti** [reply]
To avoid the impact we could define an alias named "Node" and deprecate it: in this way all existing projects would compile and they would be encouraged to migrate

**Reply by alessiostalla** [reply]
OK that can work I guess

**Comment by alessiostalla** [comment]
Another naming inconsistency that we may want to address is in transpilation traces where we have "target" result but "generated" code. IMHO we should use "generated" for both, or "target" for both.

**Reply by ftomassetti** [reply]
Should we go for "generated" then?

**Reply by alessiostalla** [reply]
That would be ideal for me
